src/libraries/System.IO.FileSystem/src/System/IO/Enumeration/FileSystemEntry.Unix.cs

```diff
@@ -133,17 +127,50 @@ public ReadOnlySpan<char> FileName
 
         // Windows never fails getting attributes, length, or time as that information comes back
         // with the native enumeration struct. As such we must not throw here.
-
         public FileAttributes Attributes
-            // It would be hard to rationalize if the attributes change after our initial find.
-            => _initialAttributes | (_status.IsReadOnly(FullPath, continueOnError: true) ? FileAttributes.ReadOnly : 0);
+        {
+            get
+            {
+                FileAttributes attributes = _initialAttributes;
+
+                if (attributes == default)
+                {
+                    if (_status.IsReadOnly(FullPath, continueOnError: true))
+                    {
+                        attributes |= FileAttributes.ReadOnly;
+                    }
+                    if (_directoryEntry.Name[0] == '.' || _status.IsHidden(FullPath, continueOnError: true))
+                    {
+                        attributes |= FileAttributes.Hidden;
+                    }
+                    if (_status.IsSymbolicLink(FullPath, continueOnError: true))
+                    {
+                        attributes |= FileAttributes.ReparsePoint;
+                    }
+                    if (_status.InitiallyDirectory || _status.IsDirectory(FullPath, continueOnError: true))
+                    {
+                        attributes |= FileAttributes.Directory;
+                    }
+
+                    // It would be hard to rationalize if the attributes change after our initial find
+                    if (attributes == default)
+                    {
+                        attributes = FileAttributes.Normal;
+                    }
+                }
+                return attributes;
+            }
```

Comment:
On a second thought, I think I will simplify this code by directly calling `_status.GetAttributes()`. That method should not hit the disk if the caches have been initialized already. The user would have to call `Refresh` before calling `Attributes` if they want the latest data. The `Attributes` property, as it is right now, would not retrieve anything if `Initialize` ends up changing the `default` value.